### PR TITLE
add config-spec-hash annotation to manifestwork

### PR DESCRIPTION
**Pull request by haoqing0110**
Ref: https://github.com/open-cluster-management-io/OCM/issues/114

### CODE REVIEW

File: pkg/addonmanager/controllers/agentdeploy/utils.go [hunk 1]
```diff
@@ -492,3 +499,32 @@ func getDeletionOrphaningRule(obj runtime.Object) (*workapiv1.OrphaningRule, err
...
+	}
+
+	annotations := make(map[string]string, len(configReference))
+	for _, v := range configReference {
+		if v.DesiredConfig == nil {
+			continue
+		}
+		resourceStr := v.Resource
+		if len(v.Group) > 0 {
+			resourceStr += fmt.Sprintf(".%s", v.Group)
```
Comment (qiujian16) [inline pkg/addonmanager/controllers/agentdeploy/utils.go:524]: the function needs a doc on what annotation will be generated.

Comment (haoqing0110) [reply pkg/addonmanager/controllers/agentdeploy/utils.go:524]: add some comments for this function.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**